Tetris_formal.py

```diff
@@ -0,0 +1,668 @@
+        
+            
+    
+    
+    
+    
+
+def move_left_or_right(left_or_right):
+    global temp_list
+     
+    ######### 先檢查
+    if check_left_or_right(left_or_right) :
+        
+        
+        ######### 檢驗是往左還往右
+        if left_or_right == 'left' :
+            change_in_x = -40
+    
+        elif left_or_right == 'right':
+            change_in_x = 40
+            
+            
+        ######### 這裡有設置邊界 
+        for temp in temp_list :
+            if 0 <= temp[1] + change_in_x <= 360  :
+                flag = True
+            
+            else :
+                flag = False
+                break
+            
+        if flag :
+            for temp in temp_list : 
+                temp[1] += change_in_x
+        
+    
+                
+  
+
+
+
+def move_down() :
+   
+    global temp_list
+    
+    
+    if check_stockpile()  :
+        
+        for temp in temp_list :
+            temp[2] += 40
+  
+    else :
+        
+        block_list.extend(temp_list) 
+        
+        ######### 重新生成新方塊
+        temp_list = Get_value.get_inital_value()
+        
+        
+        
+       
+######### 分隔線
+def draw_lines() :
+    
+    for x in range(0,400,40) :
+        pygame.draw.line(background, (253,253,253), (x,0), (x,800), 1)
+    
+    for y in range(0,800,40) :
+        pygame.draw.line(background, (253,253,253), (0,y), (400,y), 1)
+            
+   
+    
+    
+    
+    
+######### 放在這裡的原因為：pygame.event.get()只要用過那個事件，那個事件就會被刪除，所以要透過def，每次使用才會重新更新所有資訊
+def press_which_side() :
+    
+
+    ######### 再次讀取事件
+    for event in pygame.event.get(): 
+
+                ######### 如果點視窗的叉叉
+                if event.type == pygame.QUIT: 
+                                      
+                    ######### 離開pygame
+                    pygame.quit()  
+                                       
+                    ######### 關閉程式
+                    sys.exit()  
+                    
+                    
+                if event.type == pygame.KEYDOWN:
+                    
+                                        
+                    if event.key == pygame.K_LEFT:
+                       
+                        return 'left'
+                         
+                            
+                    elif event.key == pygame.K_RIGHT:
+                        
+                        return 'right'
+                    
+                    elif event.key == pygame.K_DOWN:
+                        
+                        return 'down'
+                        
+                    
+                    ######### 旋轉
+                    elif event.key == pygame.K_1 :
+                        
+                        return 'rotate clockwise'
+                        
+
+                    elif event.key == pygame.K_2 :
+                        
+                        return 'rotate counterclockwise'
+                        
+
+
+######### 如果堆到頂就輸了
+######### 之後就困在這個畫面
+def lose_the_game() :
```

Comment:
這個可以直接跟原本的while loop合併在一起

建議不要再寫一次while迴圈

可以改成拿到輸遊戲的畫面
```suggestion
def get_losing_scene() :
```

然後在下面的while loop判斷要顯示 `正常遊戲畫面` 還是 `輸遊戲的畫面`